drivers/serial/uart_aesc.c

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2025 aesc silicon
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT aesc_uart
+
+#include <errno.h>
+#include <soc.h>
+
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/uart.h>
+#include <zephyr/init.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/aesc/ip_identification.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(aesc_uart, CONFIG_UART_LOG_LEVEL);
+
+#define DEV_UART_IRQ_TX_EN		(1 << 0)
+#define DEV_UART_IRQ_RX_EN		(1 << 1)
+
+struct uart_aesc_data {
+	DEVICE_MMIO_NAMED_RAM(regs);
```

Comment:
Will this driver be present on SoC with an MMU? Otherwise, this is not needed.

Reply:
Not yet but eventually in the future. So, if it's not a problem I would keep it.